templates/types/streaming/express/src/controllers/llamaindex-stream.ts

```diff
@@ -20,50 +22,73 @@ type LlamaIndexResponse =
   | AgentStreamChatResponse<ToolCallLLMMessageOptions>
   | Response;
 
-export type DataParserOptions = {
-  imageUrl?: string;
-  csvFiles?: CsvFile[];
-};
-
 export const convertMessageContent = (
-  textMessage: string,
-  additionalData?: DataParserOptions,
+  content: string,
+  annotations?: JSONValue[],
 ): MessageContent => {
-  if (!additionalData) return textMessage;
-  const content: MessageContent = [
+  if (!annotations) return content;
+  return [
     {
       type: "text",
-      text: textMessage,
+      text: content,
     },
+    ...convertAnnotations(annotations),
   ];
-  if (additionalData?.imageUrl) {
-    content.push({
-      type: "image_url",
-      image_url: {
-        url: additionalData?.imageUrl,
-      },
-    });
-  }
+};
 
-  if (additionalData?.csvFiles?.length) {
-    const rawContents = additionalData.csvFiles.map((csv) => {
-      return "```csv\n" + csv.content + "\n```";
-    });
-    const csvContent =
-      "Use data from following CSV raw contents:\n" + rawContents.join("\n\n");
-    content.push({
-      type: "text",
-      text: `${csvContent}\n\n${textMessage}`,
-    });
-  }
+const convertAnnotations = (
+  annotations: JSONValue[],
+): MessageContentDetail[] => {
+  const content: MessageContentDetail[] = [];
+  annotations.forEach((annotation: JSONValue) => {
+    // first skip invalid annotation
+    if (
+      !(
+        annotation &&
```

Comment:
@thucpn happy for suggestions how to simplify this (without using `any` :D )

Reply:
might be a bit overkill for now, but please consider using it in a next PR if we're using more annotations